### ISSUE TITLE
VERY strange request

### ISSUE THREAD

**Issue by Hardcore-fs** [ForkIssues/Tracker#961]
gitlab CE
allows you to "import" a github or git lab project & make changes locally

BUT!! 
what it will not allow on the "basic" version is for you to  "Re-pull" changes since you download the master, you can push... but that is useless if you don't own the project.

Can you add an option that will allow you to "repull from some " real base project from a local client.
Basically like this:
allow a text box where we can enter a url to the original  remore-remote url, so that the local project can be synced to that, so that all this can be pushed back from the client to the "local" remote.

I can implement this currently by going into the "git config" file in a project & changing the "remote" url to the off site  git server, doing a pull.. then changing the url back to my "local" remote & pushing.

so my local client might have :

```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
        ignorecase = true
        precomposeunicode = true
[remote "origin"]
        url = ssh://git@192.168.55.1:30001/rust_oxide/opensource/bob.git
        fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
        remote = origin
        merge = refs/heads/master
```

I edit it to this:
```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
        ignorecase = true
        precomposeunicode = true
[remote "origin"]
        url = https://github.com/bobrepos/bob.git
        fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
        remote = origin
        merge = refs/heads/master

```
do a pull to get the upstream changes
re-edit it back to this to get the upstream into my personal remote server....

```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
        ignorecase = true
        precomposeunicode = true
[remote "origin"]
        url = ssh://git@192.168.55.1:30001/rust_oxide/opensource/bob.git
        fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
        remote = origin
        merge = refs/heads/master
```
I know it's long winded.... but there are actually quite a few users that would apreciate this functionality!!

**Comment by DanPristupov** [comment]
Looks like you are trying to pull changes from a repo you don't have a write access to.

Why don't you create another remote and fetch its changes regularly?

1. Right click on remotes on the sidebar. Select Add
2. Enter `https://github.com/bobrepos/bob.git`, name it `upstream`.
3. Fetch
4. Pull `upstream/master`


>  there are actually quite a few users that would apreciate this functionality!!

We should remember that the rest (`∞ - 2`) will appreciate if we keep UI clean.

**Comment by Hardcore-fs** [comment]
Because you CANNOT......  if using gitlab-PE(FOSS)... as your " local remote server".
the setup is:
Very remote source server_01->
Very remote source server_02->
Very remote source server_03->   local remote-> developer computer.

They disabled it.....yep.. I know.....

https://docs.gitlab.com/ee/user/project/repository/repository_mirroring.html
"Pull: for mirroring a repository from another location to GitLab. ".. there is a tiny little "?" at the end of the  statement".

and the people requesting it:
https://gitlab.com/gitlab-org/gitlab-foss/-/issues/48981.. 

like many others.. .I only found out about this AFTER we installed & built systems only to have the functionality turned off!!!!
